src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -118,6 +119,15 @@ public override string ToString()
 
             return m_toString;
         }
+
+        public override bool Equals(object? obj) =>
+            object.ReferenceEquals(this, obj) ||
+            (MetadataUpdater.IsSupported &&
+                obj is RuntimeConstructorInfo ci &&
+                MetadataToken == ci.MetadataToken &&
+                m_declaringType.Equals(ci.m_declaringType));
+
+        public override int GetHashCode() => ValueType.GetHashCodeOfPtr(m_handle);
```

Comment:
You should merge from main to resolve merge conflicts and fix build breaks. `GetHashCodeOfPtr` was moved recently.